Tests fail on Postgres 17

While attempting to execute tests locally I discovered an issue on Postgres 17. Tests pass against pgstac on Postgres 16 but fail against pgstac on Postgres 17.

[This repo](https://github.com/captaincoordinates/titiler-pgstac-chunker-function-issue) provides a reproduction case and an explanation of the issue, partially duplicated below:

---

This project's CI currently [uses](https://github.com/stac-utils/titiler-pgstac/blob/1d6ef40698f86e4a12b071fd5629b7494abe837d/.github/workflows/ci.yml#L38) Postgres 16, which explains why this issue has not previously been apparent.

## Scope
I was unable to replicate the issue in normal titiler-pgstac use or [this](https://github.com/captaincoordinates/pgstac-chunker-function-issue) pgstac-only repo. I suspect this could indicate an issue with the [pytest-postgresql](https://pypi.org/project/pytest-postgresql/) package used to support automated tests.

## Impact
As this issue does not appear to directly affect titiler-pgstac use its impact is likely low, however by generating false negatives in testing it can negatively impact development efforts that extend titiler-pgstac.

## Investigation
The cause appears to be within the [pgstac.chunker](https://github.com/stac-utils/pgstac/blob/497625a1ec77a197f6f2ff3e1dd7b9456bb1b3a1/src/pgstac/sql/004_search.sql#L2) function. This function joins to the pgstac.partition_steps materialized view using the EXPLAIN JSON's Relation Name field. pgstac.partition_steps has schema-prefixed table names and while Postgres 16's Relation Name field also has schema-prefixed table names, Postgres 17's does not. In Postgres 17 this creates a 0-row join product, which means calls to pgstac.partition_queries that include a datetime ORDER BY clause return 0 SQL statements, which means functions like pgstac.geojsonsearch erroneously return 0 features when they should return >0 features.

ah sorry I didn't see the readme. 

Going to have a look later this week. I'm not sure to understand everything but to me it seems the issue is on pgstac side. 

My comment about pgstac tests still stand 

when you do https://github.com/captaincoordinates/titiler-pgstac-chunker-function-issue/blob/758e559b0f291f51887aeda25d1a855f1803a42b/scripts/test.sh#L26 we will run the python test with 3 different version of the pgstac database `0.7`, `0.8` and `0.9` (but using pypgstac 0.9) 


```
tests/test_collections.py::test_assets_for_point_collections[0.7.10] Running to PgSTAC migration...
PgSTAC version: 0.7.10
FAILED

============================================================================ FAILURES ============================================================================
___________________________________________________________ test_assets_for_point_collections[0.7.10] ____________________________________________________________

app = <starlette.testclient.TestClient object at 0xffff9ce18a50>

    def test_assets_for_point_collections(app):
        """Get assets for a Point."""
        response = app.get(f"/collections/{collection_id}/point/-85.5,36.1624/assets")
        assert response.status_code == 200
        resp = response.json()
>       assert len(resp) == 2
E       assert 0 == 2
E        +  where 0 = len([])

tests/test_collections.py:22: AssertionError
```

☝️ you can see the tests fails when testing titiler-pgstac with `pgstac==0.7.10` 

if you remove `-x` in https://github.com/captaincoordinates/titiler-pgstac-chunker-function-issue/blob/758e559b0f291f51887aeda25d1a855f1803a42b/scripts/test.sh#L26 you'll see that tests passe for pgstac==0.9.8 

So going back again to my first comment: 

> I believe the tests only fails for old PgSTAC versions.
> I think the best way forward would be to drop support for pgstac <0.9

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+* disable tests for pre-0.9 pgstac versions
+
 ## 2.0.0 (2026-01-13)
 
 * change environment variable names for Postgres database connection **breaking change**

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -75,8 +75,6 @@ def load_json(filepath: str):
 
 @pytest.fixture(
     params=[
-        "0.7.10",
-        "0.8.5",
         "0.9.8",
     ],
     scope="session",

